### ISSUE TITLE
[Training] Support for 2 fwd graphs

### ISSUE THREAD

**Issue by vladimirjovanovicTT** [tenstorrent/tt-forge-fe#992]
When training, we need 2 forward passes in case layers (like BatchNorm, Dropout and some lesser known layers) have different behavior dependent on whether the model is in training mode or not.
Support for 2 forward graphs (compilation, switching) needs to be added to FFE.

This issue will be updated with API details, to be discussed offline.
